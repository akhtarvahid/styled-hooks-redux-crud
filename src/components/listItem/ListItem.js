import React from "react";
import { Table, OverlayTrigger, Tooltip } from "react-bootstrap";
import { FaEdit, FaBook } from "react-icons/fa";
import { MdDelete } from "react-icons/md";
import { GrTooltip } from "react-icons/gr";

import './listItem.scss';

function ListItem({ lists, onHandleEdit, onDelete, onChecked,
  AnyChecked, handleAllChecked, deleteAll }) {

  const renderTooltip = (list) => {
    return (
      <>
        <span>{list.title.length > 15 ?
          <OverlayTrigger
            placement="top"
            delay={{ show: 250, hide: 400 }}
            overlay={
              <Tooltip>
                {list.title}
              </Tooltip>
            }>
            <span className="tooltip-text">
              {list.title.substring(0, 15).trim() + '...'}
              <GrTooltip />
            </span>
          </OverlayTrigger>
          : list.title}
        </span>
      </>
    )
  }

  const isCheckedMoreThanOne = lists.filter(list => list.completed)?.length <= 1;

  return (
    <>
      <Table responsive className="table"
        style={{ display: lists.length < 1 && 'none' }}>
        <thead>
          <tr>
            <th><FaBook /></th>
            <th>Title</th>
            <th className="remove">Description</th>
            <th>
              <span className="action">Action</span>
              <span className={AnyChecked ? 'check-all action-border' : 'check-all'}>
                {AnyChecked && <span>All</span>}
                <input type="checkbox" onChange={handleAllChecked}
                  checked={AnyChecked} />
                {AnyChecked && <MdDelete className="delete" onClick={deleteAll}
                alt="delete"/>}
              </span>
            </th>
          </tr>
        </thead>
        <tbody>
          {lists &&
            lists.map((list) => (
              <tr key={list.id} className={list.completed ? 'selected' : ''}>
                <td><FaBook /></td>
                <td>{renderTooltip(list)}</td>
                <td className="remove">{list.description.length > 20 ? list.description.substring(0, 20).trim() + '...' : list.description}</td>
                <td className="table_edit">
                  <div>
                    {isCheckedMoreThanOne && <FaEdit onClick={() => onHandleEdit(list)} />}
                    <input key={list.id} onChange={onChecked} type="checkbox" checked={list.completed} value={list.title} />
                    {list.completed && <MdDelete className="delete" onClick={() => onDelete(list)} alt="delete"/>}
                  </div>
                </td>
              </tr>
            ))}
        </tbody>
      </Table>
      <div className="table_no-content">
        {lists.length < 1 &&
          'All Items are deleted'
        }
      </div>
    </>
  );
}

export default ListItem;