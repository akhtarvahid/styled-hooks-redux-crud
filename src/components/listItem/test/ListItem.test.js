import React from 'react';
import renderer from 'react-test-renderer';
import ListItem from '../ListItem';

test('should render ListItem Component', () => {
    let fakeProps = [{
        id: 1,
        title: "title",
        description: "description",
        completed: false
    }];
    const tree = renderer
        .create(<ListItem lists={fakeProps} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
