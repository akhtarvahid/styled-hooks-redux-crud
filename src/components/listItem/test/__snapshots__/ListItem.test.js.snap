// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ListItem Component 1`] = `
Array [
  <div
    className="table-responsive"
  >
    <table
      className="table table"
      style={
        Object {
          "display": false,
        }
      }
    >
      <thead>
        <tr>
          <th>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
              />
            </svg>
          </th>
          <th>
            Title
          </th>
          <th
            className="remove"
          >
            Description
          </th>
          <th>
            <span
              className="action"
            >
              Action
            </span>
            <span
              className="check-all"
            >
              <input
                type="checkbox"
              />
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className=""
        >
          <td>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
              />
            </svg>
          </td>
          <td>
            <span>
              title
            </span>
          </td>
          <td
            className="remove"
          >
            description
          </td>
          <td
            className="table_edit"
          >
            <div>
              <svg
                fill="currentColor"
                height="1em"
                onClick={[Function]}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                />
              </svg>
              <input
                checked={false}
                type="checkbox"
                value="title"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="table_no-content"
  />,
]
`;
